### PR TITLE
PR: Update gitignore, gitattributes, license, authors and security policy

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
Adds/updates various meta-files and syncs with with other Spyder projects; final bit of general maintenance for v2.0.0. 

Namely:

* [x] Update gitignore to modern standardized version used on other Spyder repos, with QtPy-unique additions merged in
* [x] Add gitattributes from the current standard version from the other Spyder projects, with a few minor tweaks
* [x] Update license with correct copyright line, in line with that on main Spyder repo and elsewhere, and fix EOL
* [x] Revise AUTHORS to more closely follow the standard format used on other Spyder repos, and include names and links to the main projects that become QtPy and their authors (from the README), as well as the current maintainer
* [x] Add a baseline, standardized security policy (which GitHub strongly recommends and will include in their UI), since QtPy is a very important part of the dependency chain for tens of thousands of projects and one of the top ≈200 packages on PyPI

### CODE REVIEW

File: .gitattributes
Comment (dalthviz) [inline .gitattributes:228]: Is this `Documents` section necessary here?

File: .gitattributes
Comment (dalthviz) [inline .gitattributes:209]: Is this `Data` section necessary here?

File: .gitattributes
Comment (dalthviz) [inline .gitattributes:168]: All of these extentions are necessary?

File: .gitattributes
Comment (dalthviz) [inline .gitattributes:69]: Is this `Web` section necessary here?

File: .gitattributes
Comment (dalthviz) [inline .gitattributes:95]: Is this `Plain text data` section necessary here?

File: .gitattributes
Comment (dalthviz) [inline .gitattributes:252]: Is this `Graphics` section necessary here?

File: .gitattributes
Comment (dalthviz) [inline .gitattributes:261]: Is this `Fonts` section necessary here?

File: .gitignore
Comment (dalthviz) [inline .gitignore:5]: I think we should keep this

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:2]: Its still there, just moved

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:36]: Per all available evidence, it appears to be a common typo/mispelling of `.cache`, which was already included elsewhere in the `.gitignore` list. I browsed through the first few dozen hits for it on GitHub, and every one appeared to be an accidental mispelling of `cache`, there was no tool I could find that referred to it, and there were only a few (≈1000) exact match results on Google for it, all of which were either obvious mispellings, unrelated or names/words in other languages...all except for one, the seventh overall hit, which was for **this exact file in the QtPy repo** :laughing:

File: AUTHORS.md
Comment (CAM-Gerlach) [reply AUTHORS.md:13]: You are the lead maintainer and very deserving of the title with all your hard and diligent work here, but its completely up to you if you don't want your name there. I think you should give yourself some credit there as you are a very kind, humble person, but I also totally understand if you'd rather not. So 100% your call here, whichever you prefer.

File: SECURITY.md
Comment (CAM-Gerlach) [reply SECURITY.md:1]: :heavy_check_mark:

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:5]: Its still there, just re-organized a bit

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore]: Its there so users can uncomment in situations where the `Pipfile.lock` shouldn't be checked in to version control. However, since it is recommended that it is, and no Spyder-related project, nor any project I'm involved with, uses `pipenv` nor is likely to ever (given its serious problems and decline in maintenance and popularity in favor of better and more modern alternatives like Flit and Poetry,

**Comment by CAM-Gerlach** [comment]
Just to clarify again, the `.gitignore` and `.gitattributes` files are sourced directly from the most updated versions of the standardized `.gitignore` and `.gitattributes` files we've been using across many/most Spyder projects (including the docs, website, theme, spyder-kernels, gettext-helpers, etc) for at least three years now (as well as many other projects). They themselves are based off standard, widely-used templates, with some further additions based on the needs of various specific Spyder-related projects, and occasionally synced with the latest upstream improvements. I've also made sure to integrate the few instances of `.gitignore` patterns included in this project's `.gitignore` that weren't already included in the existing ones.

@dalthviz To address all the "is this necessary" questions, the primarily goal of this PR was to standardize on comprehensive, proven `.gitignore` and `.gitattributes` files that correctly cover a wide variety of cases and that have worked well across all our other projects for several years, while reducing the burden of a individually maintaining a patchwork of bespoke, manually-updated files unique to each project. Pragmatically, I believe spending a substantial amount of time bikeshedding over if each section and item is likely to be strictly necessary is likely to cost more time and effort than it could possibly save us, and more importantly negates the purpose of standardizing on a single common version of these files (that can be updated with a simple copy/paste, GitHub Action, or other automated tool in the future) to begin with.

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:11]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:57]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:83]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:87]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:90]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:94]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:97]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:100]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:104]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:110]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:113]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:134]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitignore
Comment (CAM-Gerlach) [reply .gitignore:17]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitattributes
Comment (CAM-Gerlach) [reply .gitattributes:275]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitattributes
Comment (CAM-Gerlach) [reply .gitattributes:228]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitattributes
Comment (CAM-Gerlach) [reply .gitattributes:209]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitattributes
Comment (CAM-Gerlach) [reply .gitattributes:168]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitattributes
Comment (CAM-Gerlach) [reply .gitattributes:69]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitattributes
Comment (CAM-Gerlach) [reply .gitattributes:95]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitattributes
Comment (CAM-Gerlach) [reply .gitattributes:252]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: .gitattributes
Comment (CAM-Gerlach) [reply .gitattributes:261]: See [my comment](https://github.com/spyder-ide/qtpy/pull/278#issuecomment-963716528)

File: AUTHORS.md
Comment (dalthviz) [reply AUTHORS.md:13]: Then maybe we should put here the people that have the possibility to do releases? So you, @ccordoba12 and me?

**Comment by dalthviz** [comment]
Thanks for the explanation @CAM-Gerlach ! Thinking about is still quite strange to me to have stuff related to things that probably will not be used here (maybe is just for me though). What do you think @ccordoba12 ?

File: AUTHORS.md
Comment (CAM-Gerlach) [reply AUTHORS.md:13]: If that's what you prefer, I'm happy to do that. You're lead maintainer and I don't have a strong preference, so I'll gladly implement whatever you think is best.

**Comment by CAM-Gerlach** [comment]
> Thinking about is still quite strange to me to have stuff related to things that probably will not be used here (maybe is just for me though). 

Perhaps, but so long as they're suitably comprehensive, standardized and well-proven, they shouldn't need to ever be looked at again (aside the very occasional update, that with standardized files, can be simple drag and drop). After it hasn't been a problem for you before on any of the other Spyder repos using these same standardized files for the past three years, has it? :)

Something I've discovered time and again along my own journey as a scientist and a programming is that just going with one standard, consistent approach that works in all probable cases is usually a lot more efficient then spending the time and mental energy thinking about which method I can get away with in each particular case, and the risk of making a mistake if I'm wrong. In this case, this is doubly true since there is no objective cost to just use the standardized, proven version and modest objective benefits, and no clear benefit to spending the considerable time, energy and risk now and in the future determining line by line and section by section whether each may be relevant.

**Comment by dalthviz** [comment]
Any feedback on this @ccordoba12 ?

File: AUTHORS.md
Comment (dalthviz) [reply AUTHORS.md:13]: Lets do that then :+1:

Comment (CAM-Gerlach) [reply AUTHORS.md:15]: Done, thanks! :+1:

**Comment by CAM-Gerlach** [comment]
Just to TL;DR my take @ccordoba12 , these are the standard versions we have in many/most of our other modern repos (with a few improvements making sure the `.gitignore` was a superset of the existing custom QtPy one), which improve diffs, EOL normalization, ignoring editor files, and text/binary handling. So my concerns are ironically for the same reasons as yours on the the `_version` file change (though in that case, I initially tried to preserve it as much as possible and only relented when it was clear it conflicted with the goal of the PR, whereas here porting the standardized version **is** the goal of the PR).

**Comment by ccordoba12** [comment]
> Any feedback on this @ccordoba12 ?

Although I find the changes in `.gitattributes` and `.gitignore` a bit of an overkill for a such a simple project as this one, I'm fine with them.

**Review by ccordoba12 — APPROVED** [review]
Thanks @CAM-Gerlach!

**Pull request merged**
